canonical-web-and-design/maas-ui
# refactor(ui): decouple KVM forms from active KVM and route params

## Done

- Decoupled KVM forms from active KVM in state and route params, so now you must provide the id manually when opening a form (if the form requires an id)
- Moved `components/PodDetailsActionMenu` to `VirshDetails/VirshDetailsHeader/VirshDetailsActionMenu` since it's only used for virsh KVMs 

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM list and click "Add KVM"
- Check that the Add KVM form opens properly
- Go to the details page of a LXD KVM and click "Add virtual machine"
- Check that you can still compose a machine
- Check that you can still open the refresh/delete/VM action forms as well
- Go to the details page of a virsh KVM and check that you can compose a machine there as well

## Fixes

Fixes canonical-web-and-design/app-squad#364
